src/axom/quest/DistributedClosestPoint.hpp

```diff
@@ -455,9 +455,16 @@ class DistributedClosestPointImpl
     axom::Array<axom::IndexType> domIds(ptCount, ptCount);
     std::size_t copiedCount = 0;
     conduit::Node tmpValues;
-    for(conduit::index_t d = 0; d < mdMeshNode.number_of_children(); ++d)
+    for(axom::IndexType d = 0; d < mdMeshNode.number_of_children(); ++d)
     {
       const conduit::Node& domain = mdMeshNode.child(d);
+
+      axom::IndexType domainId = d;
+      if(domain.has_path("state/domain_id"))
+      {
+        domainId = domain.fetch_existing("state/domain_id").to_int32();
+      }
```

Comment:
Minor: Should we add a check that the `domain_id`s are distinct? Perhaps in a `verify` function?

I can imagine the algorithm behaving badly if the domains are not distinct.